Tests/eppo/AssignmentCacheTests.swift

```diff
@@ -16,94 +15,166 @@ final class AssignmentCacheTests: XCTestCase {
+        XCTAssertFalse(cache.shouldLogAssignment(key: key1))
+    }
 
-        cache.setLastLoggedAssignment(key: key1)
-        XCTAssertTrue(cache.hasLoggedAssignment(key: key1))
-        XCTAssertFalse(cache.hasLoggedAssignment(key: key2))
+    func testShouldLogAssignmentDifferentVariations() {
+        let key1 = AssignmentCacheKey(
+            subjectKey: "user123", 
+            flagKey: "feature-flag",
+            allocationKey: "allocation1",
+            variationKey: "variation1" // Different variation
+        )
+        let key2 = AssignmentCacheKey(
+            subjectKey: "user123",
+            flagKey: "feature-flag", 
+            allocationKey: "allocation1",
+            variationKey: "variation2" // Different variation
+        )
 
-        cache.setLastLoggedAssignment(key: key2)
-        XCTAssertFalse(cache.hasLoggedAssignment(key: key1))
-        XCTAssertTrue(cache.hasLoggedAssignment(key: key2))
+        // First assignment should log and cache variation1
+        XCTAssertTrue(cache.shouldLogAssignment(key: key1))
+        
+        // Different variation should also log (overwrites cache with variation2)
+        XCTAssertTrue(cache.shouldLogAssignment(key: key2))
+        
+        // Now key1 (variation1) should log again since cache has variation2
+        XCTAssertTrue(cache.shouldLogAssignment(key: key1))
+        
+        // key1 (variation1) should not log again since cache now has variation1
+        XCTAssertFalse(cache.shouldLogAssignment(key: key1))
     }
 
-    func testConcurrentCacheAccess() async throws {
-        let iterations = 1000
-        let concurrentTasks = 10
+    // MARK: - Race Condition Tests
 
+    func testDeprecatedHasLoggedAssignmentRaceCondition() async throws {
+        // This test demonstrates why hasLoggedAssignment + setLastLoggedAssignment is unsafe
+        // and why we need the atomic shouldLogAssignment method
+        
         let key = AssignmentCacheKey(
-            subjectKey: "Math",
-            flagKey: "TestFlag",
-            allocationKey: "A1",
-            variationKey: "VariationA"
+            subjectKey: "user123",
+            flagKey: "feature-flag",
+            allocationKey: "allocation1", 
+            variationKey: "variation1"
         )
-
+        
+        let concurrentTasks = 100
+        var loggedCount = 0
+        let resultsQueue = DispatchQueue(label: "results.queue")
+        
         await withTaskGroup(of: Void.self) { group in
-            // Add reader tasks
+            // Launch many concurrent check-then-act operations using deprecated API
             for _ in 0..<concurrentTasks {
                 group.addTask {
-                    for _ in 0..<iterations {
-                        _ = self.cache.hasLoggedAssignment(key: key)
+                    // This is the UNSAFE pattern that can cause race conditions
+                    if !self.cache.hasLoggedAssignment(key: key) {  // Read
```

Comment:
Also confirmed that the deprecation message is visible in xcode

<img width="533" height="76" alt="Screenshot 2025-12-31 at 3 27 41 PM" src="https://github.com/user-attachments/assets/850cbf8b-4cdf-4087-8df3-581a0f46b69f" />